download/tests/read-proxy-env.rs

```diff
@@ -0,0 +1,50 @@
+#![cfg(feature = "reqwest-backend")]
+
+use std::env::{remove_var, set_var};
+use std::sync::Mutex;
+
+use env_proxy::for_url;
+use lazy_static::lazy_static;
+use url::Url;
+
+lazy_static! {
+    static ref LOCK: Mutex<()> = Mutex::new(());
```

Comment:
I don't think the lock is of any use right now - we have one and only one test, in one test script, and nothing else is consulting it, so it isn't going to lock anything out: just ditch it.